test/integration/targets/azure_rm_virtualmachine_facts/tasks/main.yml

```diff
@@ -0,0 +1,109 @@
+- name: Retrieve vms facts (all vms of the resource group)
+  azure_rm_virtualmachine_facts:
+    resource_group: "{{ resource_group }}"
+  register: results
+
+- name: Fail if the resource group has vms inside
+  fail:
+    msg: "The resource group must be empty of virtual machines in order to run this integration test."
```

Comment:
Why is this a requirement of the test?

Reply:
I must have control over the vms the resource group has in order to assure appropriately the module results.

Reply:
You should be able to update the tests to work even when there are additional VMs present. For example:

For the "Assert that facts module returned 2 vms" test check for the 2 VMs you created instead of any 2 VMs.

For these tests:

- Retrieve vms facts (filtering by tag)
- Retrieve vms facts (filtering by name)
- Retrieve vms facts (filtering by wrong name)

Use randomly generated tags/names to avoid conflicts with other tests.

Reply:
I think it would be better if the additional VMs present were removed in the beginning of the test, before the validations. What do you think?

Reply:
We generally try to avoid removing resources in a test that were not created by that test. Also, removing VMs isn't immediate, so would that actually solve anything without waiting for all the VMs to be completely deleted?

Reply:
I think it's important to validate the module with no filter defined, when the facts of all VMs of the resource group are retrieved. To reach that goal, I see 2 options:
- By deleting the additional VMs present before such validation
- By counting how many VMs exist and assure if after creating 2 VMs the total is +2

I prefer the first option because keeping the VMs yet there's the possibility of getting conflicts during the "filtering by tag" validation. If some previously VM has the same tested tag, the validation will fail.

Reply:
The second option is much more workable for testing. The issue with the same tag (or name) existing can be eliminated by making sure that sufficiently long random values are used.

Reply:
Ok, I’ll make the changes.